allennlp/models/simple_tagger.py

```diff
@@ -141,6 +135,16 @@ def tag(self, text_field: TextField) -> Dict[str, Any]:
         predictions = output_dict["class_probabilities"].data.squeeze(0)
         _, argmax = predictions.max(-1)
         indices = argmax.squeeze(1).numpy()
-        tags = [self.vocabulary.get_token_from_index(x, namespace="tags") for x in indices]
+        tags = [self.vocab.get_token_from_index(x, namespace="tags") for x in indices]
 
         return {"tags": tags, "class_probabilities": predictions.numpy()}
+
+    @classmethod
+    def from_params(cls, vocab: Vocabulary, params: Params) -> 'SimpleTagger':
+        hidden_size = params.pop("hidden_size", 200)
+        num_layers = params.pop("num_layers", 2)
+        token_embedder = TokenEmbedder.from_params(vocab, params.pop("token_embedder"))
+        return cls(vocab=vocab,
```

Comment:
Yeah, I was thinking:

```python
class ComplicatedTagger(SimpleTagger):
    @overrides
    @classmethod
    def from_params(cls, params: Params) -> ComplicatedTagger:
        # pop ComplicatedTagger-specific params here
        super_instance = super(ComplicatedTagger, cls).from_params(params)
        # use super_instance and params popped earlier to get whatever constructor args you need
        return cls(args)
```

In this case, having this be a class method buys you nothing.  The only time it might help you is if you subclass one of these without changing the parameters or the constructor.  That seems like a pretty rare case, especially with models (might be more likely in some of the data processing stuff, but still pretty unlikely).  I don't have any strong opinion on this point, and am fine with whatever you two think is best.